src/Core/src/Platform/ImageSourcePartWrapper.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Maui.Handlers;
+using System.Threading.Tasks;
+
+namespace Microsoft.Maui
+{
+	internal partial class ImageSourcePartWrapper<T> : IImageSourcePart
```

Comment:
This generalizes code for loading an image so that all code can benefit from cancel behavior and cases where the destination goes poof